libs/ui/src/battery_display.tsx

```diff
@@ -0,0 +1,71 @@
+import styled from 'styled-components';
+import type { BatteryInfo as BatteryInfoType } from '@votingworks/backend';
+import { format } from '@votingworks/utils';
+import { assert } from '@votingworks/basics';
+import { IconProps, Icons } from './icons';
+import { Font } from './typography';
+import { useSystemCallApi } from './system_call_api';
+
+const BatteryInfo = styled.div`
+  align-self: center;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  min-width: 2.1rem;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 4px;
+`;
+
+function getBatteryIcon(batteryInfo: BatteryInfoType): JSX.Element {
+  assert(
+    batteryInfo.level >= 0 && batteryInfo.level <= 1,
+    'Invalid battery level'
+  );
+  const quarters = Math.round(batteryInfo.level * 4);
+  const showDanger = quarters <= 1 && batteryInfo.discharging;
+  const iconProps: IconProps = {
+    color: showDanger ? 'danger' : undefined,
+    style: { fontSize: '1.2em' },
+  };
+
+  switch (quarters) {
+    case 0:
+      return <Icons.BatteryEmpty {...iconProps} />;
+    case 1:
+      return <Icons.BatteryQuarter {...iconProps} />;
+    case 2:
+      return <Icons.BatteryHalf {...iconProps} />;
+    case 3:
+      return <Icons.BatteryThreeQuarters {...iconProps} />;
+    case 4:
+      return <Icons.BatteryFull {...iconProps} />;
+    /* istanbul ignore next */
+    default:
+      throw new Error('Invalid battery level');
+  }
+}
+
+export function BatteryDisplay(): JSX.Element {
+  const systemCallApi = useSystemCallApi();
+  const batteryInfoQuery = systemCallApi.getBatteryInfo.useQuery();
+  const batteryInfo = batteryInfoQuery.data;
+
+  return (
+    <BatteryInfo>
+      <IconContainer>
+        {batteryInfo ? getBatteryIcon(batteryInfo) : <Icons.BatteryFull />}
+        {batteryInfo && !batteryInfo.discharging && (
+          <Icons.Bolt style={{ fontSize: '0.8em' }} />
+        )}
+      </IconContainer>
+      <Font weight="semiBold" style={{ fontSize: '0.8em', lineHeight: '0.9' }}>
```

Comment:
nit: seeing the use of the `style` prop a few times here, which is functionally fine, but breaks from the pattern of using styled components - non-blocking, but consider using the latter for consistency in the codebase

Reply:
I'm don't think we're being consistent currently. I've seen @jonahkagan use `style` quite a bit lately, and I understand why, because if it's only one or two properties I think it's more readable to have it inline, TailwindCSS style. I think we would have to discuss to decide what our ideal pattern is.

But I think within-file consistency makes sense - I moved the font props up into the styled component in 370f8098a, but left the `Icons.Bolt` style because it seems like a reasonable use case for inline styles - I'd rather have `fontSize` inline than make a new `styled` component or do something tricky with selectors to add the one-off property.